acis_thermal_check/state_builder.py

```diff
@@ -0,0 +1,294 @@
+        """
+        Give the LegacyStateBuilder arguments that were passed in 
+        from the command line, and set up the connection to the 
+        commanded states database
+
+        Parameters
+        ----------
+        args : ArgumentParser object
+            Command line arguments
+        """
+        # Connect to database (NEED TO USE aca_read for sybase; user is ignored for sqlite)
+        server = ('sybase' if args.cmd_states_db == 'sybase' else
+                  os.path.join(os.environ['SKA'], 'data', 'cmd_states', 'cmd_states.db3'))
+        self.logger.info('Connecting to {} to get cmd_states'.format(server))
+        self.db = Ska.DBI.DBI(dbi=args.cmd_states_db, server=server, user='aca_read',
+                              database='aca')
+        super(LegacyStateBuilder, self).set_options(args)
+        if self.args.backstop_file is not None:
+            self._get_bs_cmds()
+
+    def _get_bs_cmds(self):
+        """
+        Internal method used to btain commands from the backstop 
+        file and store them.
+        """
+        import Ska.ParseCM
+        if os.path.isdir(self.args.backstop_file):
+            backstop_file = globfile(os.path.join(self.args.backstop_file, 
+                                                  'CR*.backstop'))
+        else:
+            backstop_file = self.args.backstop_file
+        self.logger.info('Using backstop file %s' % backstop_file)
+        bs_cmds = Ska.ParseCM.read_backstop(backstop_file)
+        self.logger.info('Found %d backstop commands between %s and %s' %
+                         (len(bs_cmds), bs_cmds[0]['date'], bs_cmds[-1]['date']))
+        self.bs_cmds = bs_cmds
+        self.tstart = bs_cmds[0]['time']
+        self.tstop = bs_cmds[-1]['time']
+
+    def _set_initial_state(self, tlm):
+        """
+        Internal method used to get the initial state corresponding 
+        to the end of available telemetry (minus a bit).
+
+        The original logic in get_state0() is to return a state that 
+        is absolutely, positively reliable by insisting that the 
+        returned state is at least ``date_margin`` days old, where the 
+        default is 10 days. That is too conservative (given the way 
+        commanded states are actually managed) and not what is desired
+        here, which is a recent state from which to start thermal propagation.
+
+        Instead we supply ``date_margin=None`` so that get_state0 will 
+        find the newest state consistent with the ``date`` criterion 
+        and pcad_mode == 'NPNT'.
+
+        Parameters
+        ----------
+        tlm : dictionary
+            Dictionary containg temperature and other telemetry
+        """
+        # The -5 here has us back off from the last telemetry reading just a bit
+        start = DateTime(tlm['date'][-5])
+        state0 = cmd_states.get_state0(start.date, self.db, datepar='datestart', 
+                                       date_margin=None)
```

Comment:
Would it be useful to set date_margin explicitly back to 10 for this PR?  It seems like changing the behavior (from the default 10 to the more-reasonable None) is not within refactoring scope.

Reply:
Hi @jeanconn

This change is actually not being made in this PR—it was originally part of a PR made by @taldcroft:

https://github.com/acisops/acis_thermal_check/pull/4

So I think it’s fine.